semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -196,4 +257,257 @@ public static boolean isValidUrl(String url) {
   public static URL getNormalizedUrl(String url) throws MalformedURLException {
     return new URL(new URL(RELATIVE_URL_CONTEXT), url);
   }
+
+  public static Optional<String> getHttpUserAgent(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String useragent : USER_AGENT_ATTRIBUTES) {
+      if (attributeValueMap.get(useragent) != null
+          && !StringUtils.isEmpty(attributeValueMap.get(useragent).getValue())) {
+        return Optional.of(attributeValueMap.get(useragent).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpUserAgentFromHeader(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    String userAgentKey = RawSpanConstants.getValue(HTTP_USER_AGENT_REQUEST_HEADER);
+    if (attributeValueMap.get(userAgentKey) != null
+        && !StringUtils.isEmpty(attributeValueMap.get(userAgentKey).getValue())) {
+      return Optional.of(attributeValueMap.get(userAgentKey).getValue());
+    }
+
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpHost(Event event) {
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && isAbsoluteUrl(url.get())) {
+      try {
+        return Optional.of(getNormalizedUrl(url.get()).getAuthority());
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    }
+    return getHttpHostUtil(event);
+  }
+
+  private static Optional<String> getHttpHostUtil(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String host : HOST_ATTRIBUTES) {
+      if (attributeValueMap.get(host) != null) {
+        return Optional.of(attributeValueMap.get(host).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpPath(Event event) {
+    Optional<String> path = getHttpPathUtil(event);
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && path.isEmpty()) {
+
+      try {
+        String pathval = getNormalizedUrl(url.get()).getPath();
+        if (StringUtils.isBlank(pathval)) {
+          pathval = SLASH;
+        }
+        return Optional.of(removeTrailingSlash(pathval));
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    }
+    return path;
+  }
+
+  private static Optional<String> getHttpPathUtil(Event event) {
```

Comment:
Can we have a better name for the function? May be `getHttpPathAttribute`?

Reply:
+1 util is bit too generic

Reply:
Renamed to getHttpPathFromRawAttributes